pkg/agent/batch_remote_write_client.go

```diff
@@ -103,6 +132,7 @@ func (b *BatchWriteClient) batch(ctx context.Context) error {
 		level.Warn(b.logger).Log("msg", "batch write client failed to send profiles", "count", len(batch), "err", err)
 		return err
 	}
+	b.metrics.writeRawWithRetriesLatency.Observe(time.Since(start).Seconds())
```

Comment:
this is just nitpicking: I'd instead use this with a defer statement as soon as we capture the start time.